backend/Origam.Server/Session Stores/WorkflowSessionStore.cs

```diff
@@ -132,31 +132,34 @@ public object ExecuteAction(string actionId, IList<string> cachedFormIds)
 
             try
             {
-                switch (actionId)
+                lock (_lock)
```

Comment:
I would like to solve this for any `ExecuteAction` in any context store. I suggest to implement `ExecuteAction` in `SessionStore` with locking. And call `ExecuteActionInternal` which would be always locked then.